MediaBrowser.Api/Library/LibraryService.cs

```diff
@@ -1086,44 +1063,26 @@ private ThemeMediaResult GetThemeSongs(GetThemeSongs request)
             var item = string.IsNullOrEmpty(request.Id)
                            ? (!request.UserId.Equals(Guid.Empty)
                                   ? _libraryManager.GetUserRootFolder()
-                                  : (Folder)_libraryManager.RootFolder)
+                                  : _libraryManager.RootFolder)
                            : _libraryManager.GetItemById(request.Id);
 
             if (item == null)
             {
                 throw new ResourceNotFoundException("Item not found.");
             }
 
-            BaseItem[] themeItems = Array.Empty<BaseItem>();
+            IEnumerable<BaseItem> themeItems = item.GetThemeSongs();
 
-            while (true)
+            while (!themeItems.Any() && request.InheritFromParent && item.GetParent() != null)
             {
-                themeItems = item.GetThemeSongs().ToArray();
-
-                if (themeItems.Length > 0)
-                {
-                    break;
-                }
-
-                if (!request.InheritFromParent)
-                {
-                    break;
-                }
-
-                var parent = item.GetParent();
-                if (parent == null)
-                {
-                    break;
-                }
-                item = parent;
+                item = item.GetParent();
+                themeItems = item.GetThemeSongs();
```

Comment:
`GetParent()` can return null, this is a possible `NullRefException`

Reply:
This is checked for in the conditions for the while loop

Reply:
You already use the result in the next line...

Reply:
The loop checks if `item.GetParent()` is null, not `item`